### PR TITLE
Configure services additionally required by the master branch

### PR DESCRIPTION
**Pull request by jakzal**
| Q             | A
| ------------- | ---
| Branch?       | master 
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | Re #36647
| License       | MIT
| Doc PR        | -

Additionally run tests for:

* couchbase
* sqs
* kafka
* mongodb (although currently skipped due to #36702)

Before: Tests: 1893, Assertions: 5178, Skipped: 105.
After: Tests: 2042, Assertions: 5407, Skipped: 120.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -40,19 +40,69 @@ jobs:
...
+                    - 8092:8092
+                    - 8093:8093
+                    - 8094:8094
+                    - 11210:11210
+            sqs:
+                image: asyncaws/testing-sqs
+                ports:
+                    - 9494:9494
+            zookeeper:
+                image: wurstmeister/zookeeper:3.4.6
```
Comment (nicolas-grekas) [inline .github/workflows/tests.yml:60]: shouldn't this go to 4.4?

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -40,19 +40,69 @@ jobs:
...
+            kafka:
+                image: wurstmeister/kafka:2.12-2.4.1
+                ports:
+                    - 9092:9092
+                env:
+                    KAFKA_AUTO_CREATE_TOPICS_ENABLE: false
+                    KAFKA_CREATE_TOPICS: 'test-topic:1:1:compact'
+                    KAFKA_ADVERTISED_HOST_NAME: localhost
+                    KAFKA_ZOOKEEPER_CONNECT: 'zookeeper:2181'
```
Comment (nicolas-grekas) [inline .github/workflows/tests.yml:69]: ok, nevermind, just saw this

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -40,19 +40,69 @@ jobs:
...
+            couchbase:
+                image: couchbase:6.5.1
+                ports:
+                    - 8091:8091
+                    - 8092:8092
+                    - 8093:8093
+                    - 8094:8094
+                    - 11210:11210
+            sqs:
+                image: asyncaws/testing-sqs
```
Comment (bendavies) [inline .github/workflows/tests.yml:56]: have you considered https://github.com/localstack/localstack?

Comment (jakzal) [reply .github/workflows/tests.yml:56]: No, I'm mainly migrating what was there on Travis.

Comment (bendavies) [reply .github/workflows/tests.yml:56]: aha, fair.

Comment (jderusse) [reply .github/workflows/tests.yml:56]: Last time we tried (@Nyholm and I) the image did not match our expectations:
- does not supports all features (MessageAttributes used to store `messenger headers`)
- Api contract is not ISO with AWS (mostly format of errors/exception)

But we may have mistaken our implementation, if you know `localstack` well, up for a PR?

Comment (bendavies) [reply .github/workflows/tests.yml:56]: We use it, but haven't not encountered those issues. Your impl is probably a lot lower level than ours, if you're talking about `async-aws`!

**Comment by nicolas-grekas** [comment]
I fixed Travis on master/5.0 and created a small conflict with your PR. Can you please rebase?

**Comment by nicolas-grekas** [comment]
Removing the 2nd commit is enough.
I would have done it for you if github allowed pushing of your fork.
I'm sad they changed the default value for this checkbox :'(

**Comment by jakzal** [comment]
Removed the second commit and rebased 👍

**Comment by nicolas-grekas** [comment]
Thank you @jakzal.

**Pull request merged**